source/test/TestApp/PlayFabGroupsTest.cpp

```diff
@@ -0,0 +1,85 @@
+
+#include "TestAppPch.h"
+
+#include "PlayFabGroupsTest.h"
+#include "playfab/PlayFabApiSettings.h"
+#include "playfab/PlayFabSettings.h"
+
+using namespace PlayFab;
+
+namespace PlayFabUnit
+{
+
+    // GROUPS API
+    void PlayFabGroupsTest::GroupsApiTest(TestContext& testContext)
+    {
+        auto settings = std::make_shared<PlayFabApiSettings>();
+
+        auto req = ClientModels::LoginWithCustomIDRequest();
+        req.CustomId = PlayFabSettings::buildIdentifier;
+
+        clientApi->LoginWithCustomID(req, Callback(&PlayFabGroupsTest::GroupsTestLoginCallback), Callback(&PlayFabGroupsTest::GroupsTestLoginFailedCallback), &testContext);
+    }
+
+    void PlayFabGroupsTest::GroupsTestLoginCallback(const ClientModels::LoginResult& result, void* customData)
+    {
+        auto groupApi = std::make_shared<PlayFabGroupsInstanceAPI>(result.authenticationContext);
+        auto req = GroupsModels::CreateGroupRequest();
+        req.CustomTags = std::map<std::string, std::string>();
+
+        // TODO Bug 29786037: this map is required to be filled to not get a 500 error with the CreateGroup api call
+        req.CustomTags.insert(std::pair<std::string, std::string>("One", "Two"));
+
+        req.GroupName = GenerateUuidV4();
+        groupApi->CreateGroup(req, Callback(&PlayFabGroupsTest::GroupsTestGroupCallback), Callback(&PlayFabGroupsTest::GroupsTestLoginFailedCallback), customData);
```

Comment:
This reuses the GroupsTestLoginFailedCallback callback, which by name, and by text within that method doesn't reflect it as generic reusable. See the mirror comment below on that method.